docs/schema/conformance_and_extensions.md

```diff
@@ -13,14 +13,43 @@ To maximize the interoperability of data published using the Open Contracting Da
+Extensions to the standard may add new objects and fields to accommodate specific local requirements. An extension must not be created if it is possible to use existing terms from the standard. It must be possible to access an extension's schema and codelist files by replacing `extension.json` in the extension's URL with a file's path, e.g. `release-schema.json` or `codelists/codelist.csv`. For more information, refer to the Extension Explorer's [guidance on documenting extensions](https://extensions.open-contracting.org/en/publishers/#document).
 
 The schema for the standard by default allows for new fields, and does not fail validation of a file which contains unknown fields.
+
+Conformant extensions should respect the following guidelines:
+
+### Core fields
+
+* Extensions *should not* delete fields from the core schema
+* Extensions *should not* change the properties of fields from the core schema. If an extension desires to document further usage of a core standard field, it should do so through documentation, rather than changing the field's `description` property.
+
+### Structure
+
+* All definitions and properties *must* set a `title`, `description` and `type`, unless they are originally defined in the core schema or in another extension in which case they *must* set a `$ref` to the existing object
+* If a field's `type` is "array", `items` *must* be set
+* If using `items`, its `type` *must* only include "array", "number" and/or "string"
+* If an array field's `wholeListMerge` and `omitWhenMerged` properties are not used or are set to `false`, the object fields under it *must* have an `id` field and this `id` field *must* be required
+
+### Codelists
+
+* If `openCodelist` is `true`, `enum` *must not* be set
+* If `openCodelist` is `false`, `enum` *must* be set
+* If a field has an `enum`, this *must* be expressed as a csv codelist
+* If a field has an `enum`, `codelist` and `openCodelist` *must* be set
+* If adding codes to an existing codelist the codelist filename *must* append `+` to the start of the core codelist filename, for example `+documentType.csv`
+
+### Field and code names
+
+* Definition names *must* be UpperCamelCase
+* Field names *must* be lowerCamelCase
+* Definition and field names *must* contain only ASCII alphabetical letters
+* If an acronym is used within a field or definition name, the acronym *should* be all UPPERCASE, unless it is at the beginning of the name, in which case it *should* be all lowercase.
```

Comment:
I've taken a pass at rewording to use consistent imperative form and for strict use of the terms 'keyword', 'property' and 'definition' from JSON Schema. The previous draft used 'properties' to mean both properties and keywords in JSON Schema, and often referred to 'fields' and 'definitions', which mixes OCDS terminology with JSON Schema terminology, e.g. `tender.tenderPeriod.startDate` is a field in OCDS, but in JSON Schema `tender`, `tenderPeriod` and `startDate` are properties, and `tender` and `tenderPeriod` reference the `Tender` and `Period` definitions, respectively.

The second bullet point might need to be further refined depending on https://github.com/open-contracting/standard/issues/582#issuecomment-2484286699

```suggestion
### Changes to existing fields

A conformant extension *should not*:

* Delete properties or definitions from the OCDS schema.
* Change the value of keywords in the OCDS schema. If an extension desires to document further usage of a field in the OCDS, it should do so through documentation, rather than changing the field's `description` keyword.

### New fields

A conformant extension *must*:

* Set the `title`, `description` and `type` keywords for all new definitions and properties
* Set `$ref` for fields that reuse a definition in the OCDS schema.
* Set the `items` keyword for properties of `type` "array"
* Not include types other than "array", "number" and/or "string" in the `types` keyword under an `items` keyword.
* Ensure that any definition referenced in a property of `type` array, whose `wholeListMerge` and `omitWhenMerged` keywords are unset or set to `false`, has a required `id` property.

### Codelists

A conformant extension *must*:

* Set `enum` for properties with `openCodelist` set to `false`
* Not set `enum` for properties with `openCodelist` set to `true`
* Set `codelist` and `openCodelist` for properties that set `enum`.
* Document titles and descriptions for `enum` values as a CSV codelist.
* Append "+" to the start of the codelist filename when adding codes to an existing codelist, e.g. `+documentType.csv`.

### Naming conventions

A conformant extension *must*:

* Use lowerCamelCase for property names, e.g. `tenderPeriod`
* Use UpperCamelCase for definition names, e.g. `RelatedProcess`
* Use only ASCII alphabetical letters for property and definition names

A conformant extension *should* use UPPERCASE for acronyms within property or definition names, unless the acronym is at the beginning of the name, in which case it *should* be all lowercase.
```

Reply:
unresolved this just so that it's visible for James when he comes to review